server/controllers/popController.ts

```diff
@@ -0,0 +1,141 @@
+import { RequestHandler } from 'express'
+import getPopService from '../services/serviceInjection'
+
+export const renderIndex: RequestHandler = async (req, res, next) => {
+  try {
+    const { scenario } = req.query
+    res.render('pages/pop/index', { scenario, session: req.session })
+  } catch (error) {
+    next(error)
+  }
+}
+
+export const renderPopDetails = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const userProfile = await popService.getUserDetails(req.session.user_id)
+      res.render('pages/pop/details', { userProfile })
+    } catch (error) {
+      next(error)
+    }
+  }
+}
+
+export const renderPopProgress = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const progressData = await popService.getProgressDetails()
+      res.render('pages/pop/progress', progressData)
+    } catch (error) {
+      next(error)
+    }
+  }
+}
+
+export const renderMessageThread = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const message = await popService.getMessageById(req.params.id)
```

Comment:
Can we fake passing the user id into this call, we would expect to do this in real life.
Also, is this really getMessageThreadById?